tests/src/rules/extensions.js

```diff
@@ -151,6 +159,7 @@ ruleTester.run('extensions', rule, {
   invalid: [
     test({
       code: 'import a from "a/index.js"',
+      output: 'import a from "a/index"',
```

Comment:
this isn't a safe autofix unless we can validate that `a/index` and `a/index.js` resolve to the same thing.

if we can't add that validation using custom resolvers, then this would need to be a suggestion instead of an autofix.